src/components/configMaps/ConfigMap.tsx

```diff
@@ -233,35 +236,61 @@ export function ConfigMapForm({ id, appId, name = "", external, data = null, typ
     async function handleDelete() {
         try {
             if (!envId) {
-                const { result } = await deleteConfig(id, appId, name)
+                await deleteConfig(id, appId, name)
                 toast.success('Successfully deleted')
                 updateForm(listIndex, null)
             }
             else {
-                const { result } = await deleteEnvironmentConfig(id, appId, envId, name);
+                await deleteEnvironmentConfig(id, appId, envId, name);
                 toast.success('Successfully deleted')
                 updateForm(true)
             }
-
         }
         catch (err) {
             showError(err)
         }
     }
 
+    function handleFilePermission(e): void {
+        let permissionValue = e.target.value;
+        setFilePermissionValue({ value: permissionValue, error: "" });
+    }
+
     async function handleSubmit(e) {
+        const configmapNameRegex = new RegExp(/^[-.a-zA-Z0-9]+$/);
```

Comment:
this regex is wrong... i dont think configmap name can start with - or digit or .

Reply:
we should add validation for configmap keys also
Each key under the data or the binaryData field must consist of alphanumeric characters, -, _ or .

Reply:
regex fixed to /^[a-z][a-z0-9-.]+[a-z]$/